hooks/gen_docs.py

```diff
@@ -1,10 +1,277 @@
+        component = component_name
+        section = target_section + "-" + component + ".yaml"
+        if section in sections:
+            component_sections.append(section)
+        elif (
+            not include_inherited
+            and KPOPS_COMPONENTS_INHERITANCE_REF[component]
+            == PipelineComponent.get_component_type()
+        ):
+            section = target_section + ".yaml"
+            if section in sections:
+                component_sections.append(section)
+        elif include_inherited:
+            while component != PipelineComponent.get_component_type():
+                component = KPOPS_COMPONENTS_INHERITANCE_REF[component]
+                section = target_section + "-" + component + ".yaml"
+                if section in sections:
+                    component_sections.append(section)
+                    break
+                elif component == PipelineComponent.get_component_type():
+                    section = target_section + ".yaml"
+                    if section in sections:
+                        component_sections.append(section)
+                        break
+    component_sections.sort(key=component_section_position_in_definition)
+    return component_sections
+
+
+def concatenate_text_files(*sources: Path, target: Path) -> None:
+    """Concatenate the given files into one
+
+    :param *sources: Files to be conatenated. The order of the inputs will be
+        retained in the result
+    :param target: File to store the result in. Note that any previous contents
+        will be deleted.
+    """
+    with target.open("w+") as f:
+        for source in sources:
+            f.write(source.read_text())
+
+
+def component_section_position_in_definition(key: str) -> int:
+    """Returns a positional value for a given str if found in the list of
+    component definition sections
+
+    :param key: A value to look for in the list of sections
+    :returns: The corresponding position or 0 if not found
+    """
+    for section in SECTIONS_ORDER:
+        if key.startswith((section + "-", section + ".")):
+            return SECTIONS_ORDER.index(section)
+    return 0
+
+
+def check_for_changes_in_kpops_component_structure() -> bool:
+    """Detect changes in the hierarchy or the structure of KPOps' components
+
+    If a change is detected, a new structure yaml is written and the dependency
+    reference yamls are emptied.
+    Also prints a message if changes are detected.
+
+    :returns: True if change is found
+    """
+    try:
+        kpops_structure = load_yaml_file(PATH_DOCS_KPOPS_STRUCTURE)
+    except OSError:
+        kpops_structure = None
+    kpops_new_structure = {
+        "kpops_components_inheritance_ref": KPOPS_COMPONENTS_INHERITANCE_REF,
+        "kpops_components_fields": KPOPS_COMPONENTS_FIELDS,
+    }
+    if kpops_new_structure != kpops_structure:
+        Path(PATH_DOCS_COMPONENTS / "dependencies").mkdir(parents=True, exist_ok=True)
```

Comment:
```suggestion
        (PATH_DOCS_COMPONENTS / "dependencies").mkdir(parents=True, exist_ok=True)
```
it's already a `Path` instance